p2p_wallet/Scenes/Main/TabBarVC/TabBarVC.swift

```diff
@@ -113,3 +134,21 @@ class TabBarVC: BEPagesVC {
         setNeedsStatusBarAppearanceUpdate()
     }
 }
+
+private final class NavigationController: UINavigationController {
```

Comment:
It's not flexible solution to hide tabbar, if you want to close tabbar later or earlier than pushViewController, you will have some troubles. You don't know when you should hide TabBar, it's a designer decision. 
Show how you will not hide tabBar if it shouldn't be hidden on push.

Reply:
And you are using unclear inheritance again. You don't expect that pushViewController will hide your tabbar. Because you have an abstraction that is used overall.

Reply:
If you have any better solution, please make it, last time you break all navigation stack just for hiding tab bar. It's also not a good solution

Reply:
I think following patterns is good, but if you 100% strict to them, you would make the code more complicated. It's not the aim of paterns

Reply:
@OldCrab  @bigearsenal 

The point is that we never know when we have to show or hide the tab bar. It's also possible that we have to show tab bar after pushing some VCs without tab bar before. 

Pushing a new VC on the top of TabbarVC it's not good way to solve a problem. That's my opinion.
We have to control hiding the tabbar by sending him a signal. The VC should send this signal. 

How we send a signal is not important. We can even do it thought DI or Dependency Inversion.